fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/StarRating.kt

```diff
@@ -0,0 +1,119 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.shopping.ui
+
+import androidx.annotation.VisibleForTesting
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.theme.FirefoxTheme
+import kotlin.math.max
+import kotlin.math.min
+import kotlin.math.round
+import kotlin.math.roundToInt
+
+private const val NUM_STARS = 5
+private const val MAX_RATING = 5f
+private const val MIN_RATING = 0f
+
+/**
+ * UI for displaying star rating bar with maximum 5 stars.
+ */
+@Composable
+fun StarRating(
+    modifier: Modifier = Modifier,
+    value: Float,
+) {
+    val rating: Float = remember(value) {
+        max(min(MAX_RATING, value), MIN_RATING).roundToNearestHalf()
+    }
+    val contentDescription = contentDescription(rating = rating)
+
+    Row(
+        modifier = modifier.semantics {
+            this.contentDescription = contentDescription
+        },
+        horizontalArrangement = Arrangement.spacedBy(4.dp),
+    ) {
+        repeat(NUM_STARS) {
+            val star = if (it < rating && it + 1 > rating) {
+                R.drawable.ic_star_half
+            } else if (it < rating) {
+                R.drawable.ic_bookmark_filled
+            } else {
+                R.drawable.ic_bookmark_outline
+            }
+
+            Image(
+                painter = painterResource(id = star),
+                contentDescription = null,
+            )
+        }
+    }
+}
+
+@Composable
+private fun contentDescription(rating: Float): String {
+    val formattedRating: Number = remember(rating) { rating.removeDecimalZero() }
```

Comment:
I'm not sure we need the `remember` here, but I did see in the PR description that this chunk of logic will be moving out of the composable eventually?

Reply:
I think we do, it saves this calculation when/if recompositions happen, let's say we add some animation tomorrow in this component, even with 100s of recompositions, this `formattedRating` will never be calculated more than once

Reply:
OK, if you foresee `StarRating` having additional recomposition triggers in the future, then yeah, I think the remember makes sense in that case.